program.sh

```diff
@@ -1,17 +1,47 @@
 #!/bin/sh
 
+# 10 min interval
+INTERVAL=600
+STOP=stop
+
 download_repository() {
-	echo "swift download $1"
-	swift download $1
+    echo "swift download $1"
+    swift download $1
 }
 
 upload_repository() {
-	swift upload $1 repository/ sync/
+    swift upload $1 repository/ sync/
 } 
 
-trap 'kill -TERM $PID; wait $PID; upload_repository $1; exit 143' TERM
+watch_upload() {
```

Comment:
Could you add a comment to explain the purpose of this function?

Reply:
The purpose of this function is to upload the container with an interval of time.  
When the process who has launched this function catch a SIGTERM, it will propagate its signal the `sleep process` and upload the container before exiting.